README.md
Push.ly Automatic Updates
=========================

Include in your plugin or theme to provide automatic updates to the
users through Push.ly server.


Usage
-----

1. Register to Push.ly if you haven't already ;)

2. Put the library directory in your plugin or theme directory and include
the init file:

```php
include( get_template_directory() . '/automatic-updates/init.php' );
```

3. Add a field on your plugin settings page or your theme options
page where users will enter their email and make it save to the option
`pushly_email`, or simply use the default settings page:

```php
// Rename this function to avoid conflict with other themes or plugins
function test_admin() {
	if ( is_admin() )
		include( get_template_directory() . '/automatic-updates/admin.php' );
}
add_action( 'init', 'test_admin' );
```


Contributing
------------

Bug reports, pull requests and general feedback are always welcome!


License
-------

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.



HISTORY.md
## 0.1 (2012-07-02)

* First release


init.php
<?php
/**
 * @package Automatic_Updates
 * @version 0.1
 */

// Define Push.ly API URL
if ( ! defined( 'PUSHLY_API_URL' ) )
	define( 'PUSHLY_API_URL', 'app.push.ly/api/v1' );

// Secret key for encryption
if ( ! defined( 'PUSHLY_SECRET' ) )
	define( 'PUSHLY_SECRET', 'set in wp-content.php for the best security' );

// $pushly_version should be an array
global $pushly_version;
if ( ! is_array( $pushly_version ) )
  $pushly_version = array();

// Load version number of this instance of the library
include( dirname( __FILE__ ) . '/version.php' );

// Save the version details
$dir = dirname( __FILE__ );
$pushly_version[$dir] = $version;

/**
 * Run after theme setup to load the most recent version of the library
 *
 * @since Automatic Updates 0.1
 */
if ( ! function_exists( 'pushly_init' ) ) {
  function pushly_init() {
		global $pushly_version;
		reset( $pushly_version );
		arsort( $pushly_version );
		$recent_version_dir = key( $pushly_version );

		include( $recent_version_dir . '/updates.php' );
  }
  add_action( 'after_setup_theme', 'pushly_init' );
}



admin.php
<?php

/**
 * Add the options page to the admin menu
 *
 * @since Automatic Updates 0.1
 */
function pushly_admin_menu() {
	add_options_page( __( 'Automatic Updates Options' ), __( 'Automatic Updates' ), 'manage_options', 'pushly-options', 'pushly_options_page' );
}
add_action( 'admin_menu', 'pushly_admin_menu' );

/**
 * Whitelist the plugin options
 *
 * @since Automatic Updates 0.1
 */
function pushly_options_init() {
	register_setting( 'pushly_settings', 'pushly_email', 'pushly_email_validate' );
}
add_action( 'admin_init', 'pushly_options_init' );

/**
 * Options page content
 *
 * @since Automatic Updates 0.1
 */
function pushly_options_page() {
	pushly_ping();

	$email = get_option( 'pushly_email' );
?>
	<div class="wrap">
		<h2><?php _e( 'Automatic Updates API options' ); ?></h2>

		<form method="post" action="options.php">
			<?php settings_fields( 'pushly_settings' ); ?>

			<table class="form-table">
				<tr valign="top">
					<th scope="row"><?php _e( 'Email' ); ?></th>
					<td>
						<input type="text" name="pushly_email" value="<?php echo $email; ?>" />
					</td>
				</tr>
			</table>
			<p class="submit">
				<input type="submit" class="button-primary" value="<?php _e( 'Save Changes' ); ?>" />
			</p>
		</form>
	</div>
<?php	
}

/**
 * Validate options page input
 *
 * @since Automatic Updates 0.1
 */
function pushly_email_validate( $email ) {
	$email = sanitize_email( $email );

	if ( empty( $email ) ) {
		add_settings_error( 'pushly_email', 'invalid_email', __( 'Please enter a valid email.' ) );
		return '';
	}

	pushly_register_site();

	return $email;
}

/**
 * Show admin notice when the plugin options are empty
 *
 * @since Automatic Updates 0.1
 */
function pushly_admin_notices() {
	global $plugin_page;

	if ( 'pushly-options' == $plugin_page || get_option( 'pushly_email' ) != false )
		return;
?>
	<div id="pushly-error" class="error fade">
	<p><strong>
<?php printf(
	__( 'To receive updates for your themes and plugins, you need to enter your email address on the <a href="%s">settings page</a>.' ),
	admin_url( 'admin.php?page=pushly-options' )
) ; ?>
	</strong></p>
</div>
<?php
}
add_action( 'admin_notices', 'pushly_admin_notices' );



thememy.php
<?php
/**
 * @package ThemeMY_
 * @version 0.1
 */
/*
Plugin Name: ThemeMY!
Plugin URI: http://thememy.com/
Description: Automatic updates for your themes bought via ThemeMY!
Version: 0.1
Author: ThemeMY!
Author URI: http://thememy.com/
License: GPLv2 or later
 */

/*
This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

// Define ThemeMY! API URL
if ( ! defined( 'THEMEMY_API_URL' ) )
	define( 'THEMEMY_API_URL', 'thememy.com/api/v1' );

if ( ! defined( 'THEMEMY_SECRET' ) )
	define( 'THEMEMY_SECRET', 'set in wp-content.php for the best security' );

/**
 * Return ThemeMY! API URL
 *
 * @since ThemeMY! 0.1
 */
function thememy_api_url( $path = '' ) {
	$scheme = force_ssl_admin() || is_ssl() ? 'https' : 'http';
	return untrailingslashit( $scheme . '://' . THEMEMY_API_URL . '/' . $path );
}

/**
 * Return website email
 *
 * @since ThemeMY! 0.1
 */
function thememy_email() {
	return get_option( 'thememy_email' );
}

/**
 * Return website domain name
 *
 * @since ThemeMY! 0.1
 */
function thememy_domain_name() {
	return parse_url( site_url(), PHP_URL_HOST );
}

/**
 * Return website secret key
 *
 * @since ThemeMY! 0.1
 */
function thememy_secret_key() {
	return wp_hash( THEMEMY_SECRET );
}

/**
 * Return authentication token for ThemeMY! API
 *
 * @since ThemeMY! 0.1
 */
function thememy_get_token() {
	$email = thememy_email();
	$domain_name = thememy_domain_name();
	$secret_key = thememy_secret_key();

	$body = compact( 'email', 'domain_name', 'secret_key' );
	$url = thememy_api_url( 'tokens' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false
	);
	$response = wp_remote_post( $url, $args );

	if ( is_wp_error( $response ) ) {
		$message = sprintf( __( 'Error contacting the updates server: %s' ), $response->get_error_message() );
		return new WP_Error( 'invalid_request', $message );
	}

	$response_code = wp_remote_retrieve_response_code( $response );
	$body = json_decode( wp_remote_retrieve_body( $response ) );

	switch ( $response_code ) {
	case 400:
		if ( 20 != $body->code ) {
			$register_site = thememy_register_site( $email );
			if ( is_wp_error( $register_site ) )
				return $register_site;
		}

		return new WP_Error( 'unconfirmed_domain', __( 'Please check your email for a message to confirm your domain name.' ) );

	case 200:
		return $body->token;

	default:
		$message = sprintf( __( 'The updates server returned an invalid reponse: %s' ), wp_remote_retrieve_response_message( $response ) );
		return new WP_Error( 'invalid_response', $message );
	}
}

/**
 * Send a non blocking request with small timeout value to delete authentication token
 *
 * @since ThemeMY! 0.1
 */
function thememy_delete_token( $token ) {
	$url = thememy_api_url( "tokens/$token" );

	$args = array(
		'method' => 'DELETE',
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'timeout' => 0.01,
		'blocking' => false,
		'sslverify' => false
	);
	wp_remote_request( $url, $args );
}

/**
 * Register the site with ThemeMY!
 *
 * @since ThemeMY! 0.1
 */
function thememy_register_site() {
	$email = thememy_email();
	$domain_name = thememy_domain_name();
	$secret_key = thememy_secret_key();

	$body = compact( 'email', 'domain_name', 'secret_key' );
	$url = thememy_api_url( 'sites' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false
	);
	$response = wp_remote_post( $url, $args );

	if ( is_wp_error( $response ) ) {
		$message = sprintf( __( 'Error contacting the updates server: %s' ), $response->get_error_message() );
		return new WP_Error( 'invalid_request', $message );
	}

	$response_code = wp_remote_retrieve_response_code( $response );
	$body = json_decode( wp_remote_retrieve_body( $response ) );

	switch ( $response_code ) {
	case 400:
		$message = sprintf( __( 'One or more errors occured: %s' ), implode( ', ', $body->errors ) );
		return new WP_Error( 'remote_error', $message );

	case 200:
		return true;

	default:
		$message = sprintf( __( 'The updates server returned an invalid reponse: %s' ), wp_remote_retrieve_response_message( $response ) );
		return new WP_Error( 'invalid_response', $message );
	}
}

/**
 * Check for themes updates
 *
 * @since ThemeMY! 0.1
 */
function thememy_update_themes( $update ) {
	global $wp_version;

	if ( empty( $update->checked ) )
		return;

	$auth_token = thememy_get_token();
	if ( is_wp_error( $auth_token ) )
		return;

	$domain_name = thememy_domain_name();

	$themes = array();
	$current_theme = get_option( 'stylesheet' );
	$installed_themes = wp_get_themes();

	foreach ( (array) $installed_themes as $theme_title => $theme ) {
		$themes[ $theme->get_stylesheet() ] = array(
			'Name'       => $theme->get('Name'),
			'Title'      => $theme->get('Name'),
			'Version'    => $theme->get('Version'),
			'Author'     => $theme->get('Author'),
			'Author URI' => $theme->get('AuthorURI'),
			'Template'   => $theme->get_template(),
			'Stylesheet' => $theme->get_stylesheet(),
		);
	}

	$body = compact( 'auth_token', 'domain_name', 'themes' );

	$url = thememy_api_url( 'sites/themes' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false,
		'timeout' => ( ( defined('DOING_CRON') && DOING_CRON ) ? 30 : 3),
		'user-agent'	=> 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' )
	);

	$raw_response = wp_remote_get( $url, $args );

	if ( ! is_wp_error( $raw_response ) && 200 == wp_remote_retrieve_response_code( $raw_response ) )
		$response = json_decode( wp_remote_retrieve_body( $raw_response ), true );
	$new_update = new stdClass;

	if ( ! empty( $response ) )
		$update->response = array_merge( $update->response, $response );

	return $update;
}
add_filter( 'pre_set_site_transient_update_themes', 'thememy_update_themes' );

/**
 * When debugging, check for updates on every request
 *
 * @since ThemeMY! 0.1
 */
function thememy_update_debug() {
	if ( defined( 'THEMEMY_DEBUG' ) && THEMEMY_DEBUG )
		set_site_transient( 'update_themes', null );
}
add_action( 'init', 'thememy_update_debug' );

/**
 * Load admin page
 *
 * @since ThemeMY! 0.1
 */
function thememy_admin() {
	if ( is_admin() )
		include plugin_dir_path( __FILE__ ) . 'admin.php';
}
add_action( 'init', 'thememy_admin' );



automatic-updates.php
<?php
/**
 * @package Automatic_Updates
 * @version 0.1
 */
/*
Plugin Name: Automatic Updates
Plugin URI: http://push.ly/
Description: Automatic updates for your themes and plugins securely delivered by Push.ly automatic updates server
Version: 0.1
Author: Push.ly
Author URI: http://push.ly/
License: GPLv2 or later
 */

/*
This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

// Define Push.ly API URL
if ( ! defined( 'PUSHLY_API_URL' ) )
	define( 'PUSHLY_API_URL', 'app.push.ly/api/v1' );

if ( ! defined( 'PUSHLY_SECRET' ) )
	define( 'PUSHLY_SECRET', 'set in wp-content.php for the best security' );

/**
 * Return Push.ly API URL
 *
 * @since Automatic Updates 0.1
 */
function pushly_api_url( $path = '' ) {
	$scheme = force_ssl_admin() || is_ssl() ? 'https' : 'http';
	return untrailingslashit( $scheme . '://' . PUSHLY_API_URL . '/' . $path );
}

/**
 * Return website email
 *
 * @since Automatic Updates 0.1
 */
function pushly_email() {
	return get_option( 'pushly_email' );
}

/**
 * Return website domain name
 *
 * @since Automatic Updates 0.1
 */
function pushly_domain_name() {
	return parse_url( site_url(), PHP_URL_HOST );
}

/**
 * Return website secret key
 *
 * @since Automatic Updates 0.1
 */
function pushly_secret_key() {
	return wp_hash( PUSHLY_SECRET );
}

/**
 * Return authentication token for Push.ly API
 *
 * @since Automatic Updates 0.1
 */
function pushly_get_token() {
	if ( ! $email = pushly_email() )
		return;

	$token = get_option( 'pushly_auth_token' );

	if ( $token ) {
		if ( time() - $token['time'] < 3600 )
			return $token;

		pushly_delete_token( $token );
		delete_option( 'pushly_auth_token' );
	}

	$domain_name = pushly_domain_name();
	$secret_key = pushly_secret_key();

	$body = compact( 'email', 'domain_name', 'secret_key' );
	$url = pushly_api_url( 'tokens' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false
	);
	$response = wp_remote_post( $url, $args );

	if ( is_wp_error( $response ) ) {
		$message = sprintf( __( 'Error contacting the updates server: %s' ), $response->get_error_message() );
		return new WP_Error( 'invalid_request', $message );
	}

	$response_code = wp_remote_retrieve_response_code( $response );
	$body = json_decode( wp_remote_retrieve_body( $response ) );

	switch ( $response_code ) {
		case 400:
			return new WP_Error( 'unconfirmed_domain', __( 'Please check your email for a message to confirm your domain name.' ) );

		case 200:
			update_option( 'pushly_auth_token', $body->token );
			return $body->token;

		default:
			$message = sprintf( __( 'The updates server returned an invalid reponse: %s' ), wp_remote_retrieve_response_message( $response ) );
			return new WP_Error( 'invalid_response', $message );
	}
}

/**
 * Send a non blocking request with small timeout value to delete authentication token
 *
 * @since Automatic Updates 0.1
 */
function pushly_delete_token( $token ) {
	$url = pushly_api_url( "tokens/$token" );

	$args = array(
		'method' => 'DELETE',
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'timeout' => 0.01,
		'blocking' => false,
		'sslverify' => false
	);
	wp_remote_request( $url, $args );
}

/**
 * Register the site with Push.ly
 *
 * @since Automatic Updates 0.1
 */
function pushly_register_site() {
	if ( ! $email = pushly_email() )
		return;

	$domain_name = pushly_domain_name();
	$secret_key = pushly_secret_key();

	$body = compact( 'email', 'domain_name', 'secret_key' );
	$url = pushly_api_url( 'sites' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false
	);
	$response = wp_remote_post( $url, $args );

	if ( is_wp_error( $response ) ) {
		$message = sprintf( __( 'Error contacting the updates server: %s' ), $response->get_error_message() );
		return new WP_Error( 'invalid_request', $message );
	}

	$response_code = wp_remote_retrieve_response_code( $response );
	$body = json_decode( wp_remote_retrieve_body( $response ) );

	switch ( $response_code ) {
		case 400:
			if ( 22 != $body->code ) {
				$message = sprintf( __( 'One or more errors occured: %s' ), implode( ', ', $body->errors ) );
				return new WP_Error( 'remote_error', $message );
			}
			break;

		case 200:
			return true;

		default:
			$message = sprintf( __( 'The updates server returned an invalid reponse: %s' ), wp_remote_retrieve_response_message( $response ) );
			return new WP_Error( 'invalid_response', $message );
	}
}

/**
 * Check for plugins updates
 *
 * @since Automatic Updates 0.1
 */
function pushly_update_plugins( $update ) {
	global $wp_version;

	if ( empty( $update->checked ) )
		return $update;

	$auth_token = pushly_get_token();

	if ( is_wp_error( $auth_token ) )
		return;

	$domain_name = pushly_domain_name();

	if ( ! function_exists( 'get_plugins' ) )
		require_once( ABSPATH . 'wp-admin/includes/plugin.php' );

	$plugins = get_plugins();
	$active_plugins  = get_option( 'active_plugins', array() );

	$body = compact( 'auth_token', 'domain_name', 'plugins', 'active_plugins' );

	$url = pushly_api_url( 'plugins/update-check' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false,
		'timeout' => ( ( defined('DOING_CRON') && DOING_CRON ) ? 30 : 3),
		'user-agent'	=> 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' )
	);

	$raw_response = wp_remote_get( $url, $args );

	if ( ! is_wp_error( $raw_response ) && 200 == wp_remote_retrieve_response_code( $raw_response ) )
		$response = (array) json_decode( wp_remote_retrieve_body( $raw_response ) );

	if ( ! empty( $response ) ) {
		$update->response = array_merge( $update->response, $response );
	}

	return $update;
}
add_filter( 'pre_set_site_transient_update_plugins', 'pushly_update_plugins' );

/**
 * Check for themes updates
 *
 * @since Automatic Updates 0.1
 */
function pushly_update_themes( $update ) {
	global $wp_version;

	if ( empty( $update->checked ) )
		return $update;

	$auth_token = pushly_get_token();

	if ( is_wp_error( $auth_token ) )
		return;

	$domain_name = pushly_domain_name();

	$themes = array();
	$current_theme = get_option( 'stylesheet' );
	$installed_themes = wp_get_themes();

	foreach ( (array) $installed_themes as $theme_title => $theme ) {
		$themes[ $theme->get_stylesheet() ] = array(
			'Name'       => $theme->get('Name'),
			'Title'      => $theme->get('Name'),
			'Version'    => $theme->get('Version'),
			'Author'     => $theme->get('Author'),
			'Author URI' => $theme->get('AuthorURI'),
			'Template'   => $theme->get_template(),
			'Stylesheet' => $theme->get_stylesheet(),
		);
	}

	$body = compact( 'auth_token', 'domain_name', 'themes', 'current_theme' );

	$url = pushly_api_url( 'themes/update-check' );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false,
		'timeout' => ( ( defined('DOING_CRON') && DOING_CRON ) ? 30 : 3),
		'user-agent'	=> 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' )
	);

	$raw_response = wp_remote_get( $url, $args );

	if ( ! is_wp_error( $raw_response ) && 200 == wp_remote_retrieve_response_code( $raw_response ) )
		$response = json_decode( wp_remote_retrieve_body( $raw_response ), true );

	if ( ! empty( $response ) )
		$update->response = array_merge( $update->response, $response );

	return $update;
}
add_filter( 'pre_set_site_transient_update_themes', 'pushly_update_themes' );

/**
 * Get plugin information
 *
 * @since Automatic Updates 0.1
 */
function pushly_get_plugin_info( $def, $action, $args ) {
	global $wp_version;

	if ( empty( $args->slug ) )
		return $def;

	$auth_token = pushly_get_token();

	if ( is_wp_error( $auth_token ) )
		return $def;

	$domain_name = pushly_domain_name();

	$body = compact( 'auth_token', 'domain_name' );

	$url = pushly_api_url( "plugins/$args->slug" );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false,
		'timeout' => 15,
		'user-agent'	=> 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' )
	);
	$request = wp_remote_get( $url, $args );

	if ( is_wp_error( $request ) ) {
		$res = new WP_Error( 'plugins_api_failed', __( 'An unexpected error occurred. Something may be wrong with Push.ly or this server&#8217;s configuration.' ), $request->get_error_message() );
	} else {
		$res = json_decode( wp_remote_retrieve_body( $request ), true );
		if ( 200 != wp_remote_retrieve_response_code( $request ) )
			$res = new WP_Error('plugins_api_failed', __( 'An unexpected error occurred. Something may be wrong with Push.ly or this server&#8217;s configuration.' ), $res->message );
	}

	return empty( $res ) ? $def : (object) $res;
}
add_filter( 'plugins_api', 'pushly_get_plugin_info', 10, 3 );

/**
 * Get theme information
 *
 * @since Automatic Updates 0.1
 */
function pushly_get_theme_info( $def, $action, $args ) {
	global $wp_version;

	if ( empty( $args->slug ) )
		return $def;

	$auth_token = pushly_get_token();

	if ( is_wp_error( $auth_token ) )
		return $def;

	$domain_name = pushly_domain_name();

	$body = compact( 'auth_token', 'domain_name' );

	$url = pushly_api_url( "themes/$args->slug" );

	$args = array(
		'headers' => array(
			'Content-Type' => 'application/json'
		),
		'body' => json_encode( $body ),
		'sslverify' => false,
		'timeout' => 15,
		'user-agent'	=> 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' )
	);
	$request = wp_remote_get( $url, $args );

	if ( is_wp_error( $request ) ) {
		$res = new WP_Error( 'themes_api_failed', __( 'An unexpected error occurred. Something may be wrong with Push.ly or this server&#8217;s configuration.' ), $request->get_error_message() );
	} else {
		$res = json_decode( wp_remote_retrieve_body( $request ), true );
		if ( 200 != wp_remote_retrieve_response_code( $request ) )
			$res = new WP_Error('themes_api_failed', __( 'An unexpected error occurred. Something may be wrong with Push.ly or this server&#8217;s configuration.' ), $res );
	}

	return empty( $res ) ? $def : (object) $res;
}
add_filter( 'themes_api', 'pushly_get_theme_info', 10, 3 );

/**
 * When debugging, check for updates on every request
 *
 * @since Automatic Updates 0.1
 */
function pushly_update_debug() {
	if ( ! defined( 'PUSHLY_DEBUG' ) || ! PUSHLY_DEBUG )
		return;

	set_site_transient( 'update_plugins', null );
	set_site_transient( 'update_themes', null );
}
add_action( 'init', 'pushly_update_debug' );

/**
 * Load admin page
 *
 * @since Automatic Updates 0.1
 */
function pushly_admin() {
	if ( is_admin() )
		include plugin_dir_path( __FILE__ ) . 'admin.php';
}
add_action( 'init', 'pushly_admin' );

